src/bundle.c

```diff
@@ -1019,13 +1021,222 @@ static int install_bundles(struct list *bundles, struct list **subs, struct mani
 	return ret;
 }
 
+static int lex_sort(const void *a, const void *b)
+{
+	const char *name1 = (char *)a;
+	const char *name2 = (char *)b;
+	return strcmp(name1, name2);
+}
+
+/* Get a list of files in a directory sorted by filename
+ * with their fullpath, returns NULL on error (errno set by
+ * opendir).
+ */
+static struct list *get_dir_files_sorted(char *path)
+{
+        DIR *dir = NULL;
+        struct dirent *ent = NULL;
+        struct list *files = NULL;
+
+        dir = opendir(path);
+        if (!dir) {
+                /* caller should use errno for detecting issues */
+                return NULL;
+        }
+
+        while ((ent = readdir(dir))) {
+                char *name = NULL;
+                if (ent->d_name[0] == '.') {
+                        continue;
+                }
+                string_or_die(&name, "%s/%s", path, ent->d_name);
+                files = list_prepend_data(files, name);
+        }
+
+        closedir(dir);
+
+        return list_sort(files, lex_sort);
+}
+
+struct alias_lookup {
+	char *alias;
+	struct list *bundles;
+};
+
+/* Deep free on alias_lookup structs
+ */
+static void free_alias_lookup(void *lookup)
+{
+        struct alias_lookup *l = (struct alias_lookup *)lookup;
+        free(l->alias);
+        list_free_list_and_data(l->bundles, free);
+}
+
+/* Get bundles for a given alias but if the alias is
+ * not actually an alias, return a list with just the
+ * bundle name
+ */
+static struct list *get_alias_bundles(struct list *alias_definitions, char *alias)
+{
+        struct list *bundles = NULL;
+        struct list *iter = alias_definitions;
+
+        while (iter) {
+                struct alias_lookup *lookup = (struct alias_lookup *)iter->data;
+                iter = iter->next;
+                if (strcmp(lookup->alias, alias) == 0) {
+                        bundles = list_deep_clone_strs(lookup->bundles);
+                        break;
+                }
+        }
+        if (!bundles) {
+                bundles = list_prepend_data(bundles, strdup_or_die(alias));
+        }
+
+        return bundles;
+}
+
+/* Parse an alias definition file for a list of bundles
+ * it specifies
+ */
+static struct list *parse_alias_file(char *fullpath)
+{
+	FILE *afile = NULL;
+	struct list *alias_content = NULL;
+	char *line = NULL;
+	size_t len = 0;
+	ssize_t read = -1;
+
+	afile = fopen(fullpath, "r");
+	if (!afile) {
+		return NULL;
+	}
+
+	while ((read = getline(&line, &len, afile)) != -1) {
+		struct alias_lookup *l = NULL;
+		char *c = NULL;
+
+		/* get the alias but ignore lines without bundles */
+		c = strchr(line, '\t');
+		if (!c) {
+			continue;
+		}
+
+		/* strip the newline if it exists */
+		if (line[read - 1] == '\n') {
+			line[read - 1] = 0;
+		}
+
+		*c = 0;
+		c++;
+
+		l = calloc(1, sizeof(struct alias_lookup));
+		ON_NULL_ABORT(l);
+
+		/* get the bundles associated with the alias
+                 * note a file with multiple copies of the same
```

Comment:
You are mixing tabs and spaces in this patch.

Reply:
Erf yes horribly.

Reply:
Should be fixed.